File: python/semantic_kernel/processes/kernel_process/kernel_process_edge.py
# Copyright (c) Microsoft. All rights reserved.


from semantic_kernel.kernel_pydantic import KernelBaseModel
from semantic_kernel.processes.kernel_process.kernel_process_function_target import KernelProcessFunctionTarget
from semantic_kernel.utils.feature_stage_decorator import experimental


@experimental
class KernelProcessEdge(KernelBaseModel):
    """Represents an edge between steps."""

    source_step_id: str
    output_target: KernelProcessFunctionTarget

File: nodejs/semantic-kernel/processes/kernel-process/kernel-process-edge.ts
import type { KernelProcessFunctionTarget } from './kernel-process-function-target'

/**
 * Represents an edge between steps.
 * @experimental
 */
export class KernelProcessEdge {
  /**
   * The source step ID.
   */
  sourceStepId: string

  /**
   * The output target for this edge.
   */
  outputTarget: KernelProcessFunctionTarget

  constructor(params: { sourceStepId: string; outputTarget: KernelProcessFunctionTarget }) {
    this.sourceStepId = params.sourceStepId
    this.outputTarget = params.outputTarget
  }
}
